core/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -1447,6 +1429,34 @@ public QueuedMessage(ProtocolMessage msg, CompletionListener listener) {
         }
     }
 
+    /**
+     * Send all pending messages which are queue.
+     * Remove them from the queue once they are send successfully
+     * Spec: RTN19a
+     */
+    public void sendPendingQueueMessages() {
+        //RTN19a
+        if (pendingMessages.queue != null && !pendingMessages.queue.isEmpty()) {
+            for (final QueuedMessage queuedMessage : pendingMessages.queue) {
+                try {
+                    send(queuedMessage.msg, false, new CompletionListener() {
```

Comment:
Just calling send() is a breach of RTN19a2, isn't it? The send is going to assign it a new msgSerial, so it won't be idempotent

Reply:
(though I've just realised we do want it to assign a new msgSerial if the resume failed, bleh)